code/modules/projectiles/gun_system.dm

```diff
@@ -326,6 +338,12 @@
 	if(CHECK_BITFIELD(flags_gun_features, GUN_IS_AIMING))
 		toggle_aim_mode(user)
 
+	if(istype(attachments["muzzle"],  /obj/item/attachable/bayonet) || istype(attachments["muzzle"],  /obj/item/attachable/bayonetknife))
+		//remove the bayonetcharge component
+		var/datum/component/bayonetcharge/comp
+		comp = GetComponent(/datum/component/bayonetcharge)
+		comp.child.remove_action(comp.charger)
```

Comment:
ditto

Reply:
Ditto?
Whats a Ditto?

Reply:
ditto means same as above, this snowflake behaviour is not ok

Reply:
done and done